### PR TITLE
plat-stm32mp1: stub stm32mp13 regulators

### PR DESCRIPTION
**Pull request by etienne-lms**
Implements stubs for SCMI regulators that are to be exposed by STM32MP13 SCMI server but are not implemented yet in OP-TEE core. The drivers for these regulators (IOD SDMMC1/2 and VREFBUF) will be implemented once there is a regulator framwork in OP-TEE. In the meantime, stubbing those allows to use the platform.

### CODE REVIEW

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 1]
```diff
@@ -205,12 +207,9 @@ struct stm32_scmi_voltd scmi_voltage_domain[] = {
...
 	VOLTD_CELL(VOLTD_SCMI_USB33, VOLTD_PWR, PWR_USB33_NAME_ID, "usb33"),
-	VOLTD_CELL(VOLTD_SCMI_SDMMC1_IO, VOLTD_PWR, PWR_SDMMC1_IO_NAME_ID,
-		   "sdmmc1"),
-	VOLTD_CELL(VOLTD_SCMI_SDMMC2_IO, VOLTD_PWR, PWR_SDMMC2_IO_NAME_ID,
-		   "sdmmc2"),
-	VOLTD_CELL(VOLTD_SCMI_VREFBUF, VOLTD_PWR, PWR_VREFBUF_NAME_ID,
-		   "vrefbuf"),
+	VOLTD_CELL(VOLTD_SCMI_SDMMC1_IO, VOLTD_STUB, 0, "sdmmc1"),
+	VOLTD_CELL(VOLTD_SCMI_SDMMC2_IO, VOLTD_STUB, 0, "sdmmc2"),
+	VOLTD_CELL(VOLTD_SCMI_VREFBUF, VOLTD_STUB, 0, "vrefbuf"),
```
Comment (patrickdelaunay) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:212]: use different index to check on index ? 0 = SDMMC1/2  and 0 = VREFBUF

VOLTD_CELL(VOLTD_SCMI_VREFBUF, VOLTD_STUB, 1, "vrefbuf"),

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 2]
```diff
@@ -795,6 +794,21 @@ static int32_t pmic_set_state(struct stm32_scmi_voltd *voltd, bool enable)
 	return rc ? SCMI_GENERIC_ERROR : SCMI_SUCCESS;
 }
 
+static int32_t stub_describe_levels(struct stm32_scmi_voltd *voltd __unused,
+				    size_t start_index, long *microvolt,
+				    size_t *nb_elts)
+{
+	if (!microvolt || !*nb_elts || start_index) {
```
Comment (patrickdelaunay) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:801]: if (start_index)
		return SCMI_INVALID_PARAMETERS;
        if (!microvolt || !*nb_elts ) {

File: core/arch/arm/plat-stm32mp1/scmi_server.c [hunk 3]
```diff
@@ -795,6 +794,21 @@ static int32_t pmic_set_state(struct stm32_scmi_voltd *voltd, bool enable)
...
+		*nb_elts = 1;
+		return SCMI_SUCCESS;
+	}
+
+	microvolt[0] = 0;
```
Comment (patrickdelaunay) [inline core/arch/arm/plat-stm32mp1/scmi_server.c:806]: can check the IO level  => need to return the expected value else unexpected driver behavior.

if (!strcmp(voltd->priv_id, PWR_VREFBUF_NAME_ID)) {
    microvolt[0] = 0; /* VREFBUF not supported */
} else {
    microvolt[0] = 3300000; /* SDMMC IO = fixed at 3.3V, HighSpeed support */
}

**Comment by etienne-lms** [comment]
Comments addressed.

File: core/arch/arm/plat-stm32mp1/scmi_server.c (hunk 1 above)
Comment (patrickdelaunay) [reply core/arch/arm/plat-stm32mp1/scmi_server.c]: OK

**Comment by patrickdelaunay** [comment]
Reviewed-by: Patrick Delaunay <patrick.delaunay@foss.st.com>

**Comment by etienne-lms** [comment]
Tag applied, thanks,

@jenswi-linaro, @jforissier, could we merge this P-R before 3.22.0 tag? It's required for MP13 platforms.

**Comment by jenswi-linaro** [comment]
> Tag applied, thanks,
> 
> @jenswi-linaro, @jforissier, could we merge this P-R before 3.22.0 tag? It's required for MP13 platforms.

OK for me.

**Comment by etienne-lms** [comment]
Thanks

**Pull request merged**